apps/gitness/src/AppMFE.tsx

```diff
@@ -0,0 +1,335 @@
+                    children: [
+                      { index: true, element: <SandboxPullRequestListPage /> },
+                      { path: 'compare/:diffRefs*?', element: <CreatePullRequest /> },
+                      {
+                        path: ':pullRequestId',
+                        element: <PullRequestLayout />,
+                        children: [
+                          {
+                            index: true,
+                            element: (
+                              <PullRequestDataProvider>
+                                <PullRequestConversationPage />
+                              </PullRequestDataProvider>
+                            )
+                          },
+                          {
+                            path: 'conversation',
+                            element: (
+                              <PullRequestDataProvider>
+                                <PullRequestConversationPage />
+                              </PullRequestDataProvider>
+                            )
+                          },
+                          {
+                            path: 'commits',
+                            element: <PullRequestCommitPage />
+                          },
+                          {
+                            path: 'changes',
+                            element: (
+                              <PullRequestDataProvider>
+                                <PullRequestChanges />
+                              </PullRequestDataProvider>
+                            )
+                          }
+                        ]
+                      }
+                    ]
+                  },
+                  {
+                    path: 'settings',
+                    element: <RepoSettingsPage useTranslationStore={useTranslationStore} />,
+                    children: [
+                      {
+                        index: true,
+                        element: <RepoSettingsGeneralPageContainer />,
+                        handle: { breadcrumb: () => <span>Settings</span> }
+                      },
+                      {
+                        path: 'general',
+                        element: <RepoSettingsGeneralPageContainer />,
+                        handle: { breadcrumb: () => <span>Settings</span> }
+                      },
+                      {
+                        path: 'rules',
+                        element: <RepoSettingsGeneralPageContainer />
+                      },
+                      {
+                        path: 'webhooks',
+                        element: <WebhookListPage />
+                      }
+                    ]
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ],
+    {
+      basename: `/ng${renderUrl}`
+    }
+  )
+
+  return (
+    <div id="gitness-shadow-root">
+      <ReactShadowRoot>
+        <style>{`${styles}`}</style>
```

Comment:
🤯 👏

Reply:
is it a good idea to render a multi-MB string via React?
could we instead use DOM APIs to append the `style` element as a child to the shadow dom parent?